crypto/threads_pthread.c

```diff
@@ -52,11 +52,12 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
     }
 
     pthread_mutexattr_init(&attr);
-    #if defined(__TANDEM) && defined(_SPT_MODEL_)
-      pthread_mutexattr_setkind_np(&attr,MUTEX_RECURSIVE_NP);
-    #else
-      pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
-    #endif
+#  if defined(PTHREAD_MUTEX_ERROR_CHECK)
+    /*
+     * We don't use recursive mutexes, but try to catch errors if we do.
+     */
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK);
```

Comment:
Should we only do this in a debug build?

Reply:
Well, the idea is that we will be able to implement rwlocks for Windows as well. Since they don't support recursion we can't support it on Linux either. I'm not sure what the benefit of only doing it in debug is. IMO, we shouldn't have different behaviour in debug builds vs production builds.

Reply:
An error checking version is slower. We could set PTHREAD_MUTEX_NORMAL in a normal build.

Reply:
PTHREAD_MUTEX_DEFAULT doesn't actually need to map to any of the 3 other types, I assume that's the fastest one and should just work.

Reply:
> An error checking version is slower.

Ah. Good point. I hadn't thought of that